server/controllers/temporary-accommodation/manage/arrivalsController.ts

```diff
@@ -108,27 +118,41 @@ export default class ArrivalsController {
     return async (req: Request, res: Response) => {
       const { premisesId, roomId, bookingId } = req.params
       const callConfig = extractCallConfig(req)
-
-      const booking = await this.bookingsService.getBooking(callConfig, premisesId, bookingId)
-
-      const updateArrival: NewArrival = {
-        notes: req.body.notes,
-        arrivalDate: DateFormats.dateAndTimeInputsToIsoString(req.body, 'arrivalDate').arrivalDate,
-        expectedDepartureDate: booking.departureDate,
-        type: 'CAS3',
-      }
-
       try {
-        // INFO: this may confuse, the API is overloading the POST with a writeback of existing and new data
-        await this.arrivalService.createArrival(callConfig, premisesId, bookingId, updateArrival)
-        req.flash('success', 'Arrival updated')
-        res.redirect(paths.bookings.show({ premisesId, roomId, bookingId }))
-      } catch (err) {
-        if (err.status === 409) {
-          insertBespokeError(err, generateConflictBespokeError(err, premisesId, roomId, 'singular'))
-          insertGenericError(err, 'arrivalDate', 'conflict')
+        const error = new Error()
+        const booking = await this.bookingsService.getBooking(callConfig, premisesId, bookingId)
+
+        const updateArrival: NewArrival = {
+          notes: req.body.notes,
+          arrivalDate: DateFormats.dateAndTimeInputsToIsoString(req.body, 'arrivalDate').arrivalDate,
+          expectedDepartureDate: booking.departureDate,
+          type: 'CAS3',
         }
 
+        if (updateArrival.arrivalDate && dateIsInFuture(updateArrival.arrivalDate)) {
+          insertGenericError(error, 'arrivalDate', 'todayOrInThePast')
+          throw error
+        }
+
+        try {
```

Comment:
nesting these try/catch's makes this harder to read and doesn't effect the behaviour